v2/internal/identity/credential_provider.go

```diff
@@ -300,3 +326,15 @@ func (c *credentialProvider) getSecret(ctx context.Context, namespace string, se
 func getSecretNameFromAnnotation(credentialFrom string, resourceNamespace string) types.NamespacedName {
 	return types.NamespacedName{Namespace: resourceNamespace, Name: credentialFrom}
 }
+
+func authModeOrDefault(mode string) (config.AuthModeOption, error) {
+	if strings.EqualFold(mode, string(config.WorkloadIdentityAuthMode)) || mode == "" {
```

Comment:
I do think it's more consistent if we're case-sensitive (see my other comment), but approving as-is as not gonna block it on this.